refactor: Fix implicit value conversion in  formatPingTime

Empact opened the pull request:
Alternative to #18252, see motivation there.

This changes `CNodeStats` to handle ping timestamps as their original incoming usec `int64_t` values until the time they need to be displayed.

Empact committed:
refactor: Convert ping time from double to int64_t

Empact committed:
refactor: Convert min ping time from double to int64_t

Empact committed:
refactor: Convert ping wait time from double to int64_t

practicalswift commented:
Concept ACK

promag commented on src/rpc/net.cpp:
```diff
@@ -165,12 +165,12 @@ static UniValue getpeerinfo(const JSONRPCRequest& request)
-        if (stats.dPingWait > 0.0)
-            obj.pushKV("pingwait", stats.dPingWait);
+        if (stats.m_ping_usec > 0)
+            obj.pushKV("pingtime", stats.m_ping_usec / 1e6);
```
Should cast to double first?

Empact replied:
Done

vasild replied:
Assuming `x` is `int64_t`, then all of these are equivalent:

```
x / 1e6;
(double)x / 1e6;
((double)x) / 1e6;
x / 1000000.0;
```

promag replied:
Indeed, I tend to prefer the explicit numerator cast though.

hebasto reviewed (commented):
Concept ACK.

fanquake commented:
cc @vasild

Empact committed:
refactor: Cast ping values to double before output

Note the divisor is a floating point literal so presumably
also floating point.

promag reviewed (commented):
ACK 1891245e7318bf625bbf67aab08a79fc3e87b61d, added cast to double and also braces.

vasild commented:
ACK 1891245

I tested this with Clang 10 and [the warning](https://github.com/bitcoin/bitcoin/pull/18252#issue-382935046) is gone.

Nit: the cast to `double` added in 1891245 is not necessary (I am ok with it).

practicalswift commented:
ACK 1891245e7318bf625bbf67aab08a79fc3e87b61d -- patch looks correct

MarcoFalke commented:
Would be nice to add (functional) tests for this, since it is uncovered right now.
